packages/venia-ui/lib/components/Wishlist/WishlistButton/wishlistButton.ee.js

```diff
@@ -0,0 +1,116 @@
+import React, { Fragment, useEffect } from 'react';
+import { AlertCircle, Heart, Check } from 'react-feather';
+import { useIntl } from 'react-intl';
+
+import { useToasts } from '@magento/peregrine';
+import { useWishlistButton } from '@magento/peregrine/lib/talons/Wishlist/WishlistButton/useWishlistButton';
+
+import { mergeClasses } from '@magento/venia-ui/lib/classify';
+import Icon from '@magento/venia-ui/lib/components/Icon';
+
+import defaultClasses from './wishlistButton.css';
+import { arrayOf, bool, number, shape, string } from 'prop-types';
+import WishlistDialog from '../WishlistDialog';
+
+const ErrorIcon = <Icon src={AlertCircle} attrs={{ width: 18 }} />;
+
+const WishlistButton = props => {
+    const classes = mergeClasses(defaultClasses, props.classes);
+
+    const talonProps = useWishlistButton({ itemOptions: props.itemOptions });
+
+    const {
+        addProductError,
+        handleButtonClick,
+        handleModalClose,
+        isDisabled,
+        isItemAdded,
+        isModalOpen
+    } = talonProps;
+
+    const { formatMessage } = useIntl();
+    const [, { addToast }] = useToasts();
+
+    useEffect(() => {
+        if (addProductError) {
+            addToast({
+                type: 'error',
+                icon: ErrorIcon,
+                message: formatMessage({
+                    id: 'wishlistButton.addError',
+                    defaultMessage:
+                        'Something went wrong adding the product to your wishlist.'
+                }),
+                timeout: 5000
+            });
+        }
+    }, [addProductError, addToast, formatMessage]);
+
+    const buttonText = isItemAdded
+        ? formatMessage({
+              id: 'wishlistButton.addedText',
+              defaultMessage: 'Added to Favorites'
+          })
+        : formatMessage({
+              id: 'wishlistButton.addText',
+              defaultMessage: 'Add to Favorites'
+          });
+
+    let buttonClass;
+    let iconClass;
+
+    if (isDisabled) {
+        buttonClass = classes.wishlistButton_disabled;
+        iconClass = classes.icon_disabled;
+    } else {
+        buttonClass = classes.wishlistButton;
+        iconClass = classes.icon;
+    }
+
+    const HeartIcon = <Icon src={Heart} classes={{ icon: iconClass }} />;
+    const CheckIcon = (
+        <Icon src={Check} classes={{ icon: classes.checkIcon }} />
+    );
+
+    const iconElement = isItemAdded ? CheckIcon : HeartIcon;
+
+    return (
+        <Fragment>
+            <button
+                disabled={isDisabled}
+                type="button"
+                className={buttonClass}
+                onClick={handleButtonClick}
+            >
+                {iconElement}
+                {buttonText}
+            </button>
+            <WishlistDialog
+                isOpen={isModalOpen}
+                onClose={handleModalClose}
+                itemOptions={props.itemOptions}
+            />
```

Comment:
I think I need to talk about this on a call because I don't quite follow what you're asking.

I will say that the differences in logic were precisely why I had to split this into two components rather than one. I needed a way to use two different talons but an almost identical component.

Honestly I feel like this reads well, but it does result in some coupling/overhead required by someone modifying one or the other.

Reply:
I think it's fine. If we find ourselves having to make a bunch of changes to both components then we can look at refactoring.